plugins/ipam/ipam.go

```diff
@@ -323,15 +325,23 @@ func (i *IPAM) NodeIPAddress(nodeID uint32) (net.IP, *net.IPNet, error) {
 	if err != nil {
 		return net.IP{}, nil, err
 	}
-	maskSize, _ := i.nodeInterconnectSubnet.Mask.Size()
-	mask := net.CIDRMask(maskSize, 32)
+	maskSize, bits := i.nodeInterconnectSubnet.Mask.Size()
+
+	mask := net.CIDRMask(maskSize, bits)
+	ip := make([]byte, len(nodeIP))
+	copy(ip, nodeIP)
 	nodeIPNetwork := &net.IPNet{
-		IP:   newIP(nodeIP).Mask(mask),
+		IP:   net.IP(ip).Mask(mask),
 		Mask: mask,
 	}
 	return nodeIP, nodeIPNetwork, nil
 }
 
+// IsNodeInterconnectIPv6 returns true if nodeInterconnectSubnet is IPv6 range
+func (i *IPAM) IsNodeInterconnectIPv6() bool {
+	return strings.Contains(i.ContivConf.GetIPAMConfig().NodeInterconnectCIDR.String(), ":")
+}
```

Comment:
Don't forget also about ContivCIDR, which IPAM may use to calculate its own node interconnect CIDR (makes sense for IPv6?), i.e. perhaps use instead: 
`strings.Contains(i.nodeInterconnectSubnet.String(), ":")`

Reply:
ContivCIDR does not work with IPv6 at this point, the IPAM plugin needs to be adapted